geode-core/src/test/java/org/apache/geode/distributed/internal/DistributionTest.java

```diff
@@ -204,4 +207,30 @@ public void testSendToEmptyListIsRejected() throws Exception {
     distribution.send(emptyList, m);
     verify(membership, never()).send(any(), any());
   }
+
+  @Test
+  public void testExceptionNestedOnStartConfigError() throws Exception {
+    Throwable cause = new RuntimeException("Exception cause");
+    Throwable exception = new MembershipConfigurationException("Test exception", cause);
+    doThrow(exception).when(membership).start();
+    try {
```

Comment:
JUnit has [ExpectedException](https://junit.org/junit4/javadoc/4.12/org/junit/rules/ExpectedException.html) which I think works better than `try` `catch` statements.  There are some examples in the Geode code base if you want some inspiration

Reply:
Thanks for the comment! TBH I’ve never used JUnit or mockito before, so any comments are welcome!

Reply:
Always glad to provide new learning opportunities for folks 😄
Plus no time like the present to learn new things!

Reply:
AssertJ also has a very convenient `assertThatThrownBy()` which lets you assert on the type of exception thrown as well as specific messages, causes, etc. There are numerous examples throughout the test code.

Reply:
Finally went for AssertJ choice. Now the test is more more clear. Thanks!